### ISSUE TITLE
can not run elasticsearch as root

### ISSUE THREAD

**Issue by oferabr** [alexcojocaru/elasticsearch-maven-plugin#68]
Hi,

I wrote some tests using the plugin, all works as expected.
when running our project on our jenkins we get "can not run elasticsearch as root".

I know that ES cannot run as root user, is there any option in the plugin settings to run it as different user? or any other solution?

Thanks

**Comment by alexcojocaru** [comment]
I did some basic research into how to run a process as a different user and haven't found anything that's not a hack or a convoluted workaround.
IF it could be done and I implement support for this, off the top I foresee some complications:
* before starting the ES process, the plugin should change the owner of the ES directory to the other user
* after terminating the ES process, it should change the owner back to the original user; there could be cases (eg. the plugin crashes, ES crashes, etc) when changing the owner back is not done, leaving the ES directory owned by the other user, and mvn clean would error out because of that
In summary, the implementation would be very prone to issues and I would rather not go forward with it.

Now, going back to the reason of your ask: I strongly suggest you look into your Jenkins configuration and  have the Jenkins process execute as a regular user. The rule of thumb is that the root user should be used sparingly.

**Comment by oferabr** [comment]
Thank you 👍

**Comment by alexcojocaru** [comment]
@oferabr  https://github.com/alexcojocaru/elasticsearch-maven-plugin/issues/72 is a duplicate of this. I am considering implementing support for it and I would appreciate your input on it.
